### ISSUE TITLE
[DOC]: Clarification about how metrics work on `compare_models` and `create_models` for cross-validation

### ISSUE THREAD

**Issue by royzawadzki** [pycaret/pycaret#3567]
### pycaret version checks

- [X] I have checked that the issue still exists on the latest versions of the docs [here](https://pycaret.gitbook.io/docs/)

### Location of the documentation

https://pycaret.readthedocs.io/en/stable/api/classification.html

### Documentation problem

Suppose I pass in a dataset to `setup` and it automatically partitions it into a train and test set. Then, I run `compare_models` and, for each model, it will compute a cross-validated metric of interest. What is not clear in the documentation is which datasets the cross-validation is performed on:
1. Is it simply the training set? 
2. Is it the whole data (training and test)? **This is my current assumption**

The option `cross_validation` makes it somewhat clear that the if you set it to `False`, then the metrics are on the hold-out set, which I assume is the same as the test set partitioned in the `setup` section.

A similar confusion arises when we are doing `create_model` and it shows the cross validation scores.

### Suggested fix for documentation

Clarify the point whether the cross validated metrics were computed on all the data or the training.
